src/os/bluestore/BlueFS.cc

```diff
@@ -3495,7 +3789,28 @@ void BlueFS::flush_range(FileWriter *h, uint64_t offset, uint64_t length)/*_WF*/
 {
   _maybe_check_vselector_LNF();
   std::unique_lock hl(h->lock);
-  _flush_range_F(h, offset, length);
+  if (h->file->is_new_wal()) {
+    //For wal files disregard offset and length and just flush current envelope.
+    _flush_wal_F(h);
+  } else {
+    _flush_range_F(h, offset, length);
+  }
+}
+
+int BlueFS::_flush_wal_F(FileWriter *h)
+{
+  ceph_assert(ceph_mutex_is_locked(h->lock));
+  ceph_assert(h->file->is_new_wal());
+  uint64_t content_length = h->get_buffer_length() - sizeof(File::wal_flush_t::wal_length_t);
+  h->append((char*)&h->file->wal_marker.v[0], 8);
```

Comment:
Please replace 8 with the relevant sizeof or even tail_size() call.

Reply:
Done.